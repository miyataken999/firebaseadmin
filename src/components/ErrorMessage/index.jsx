import React from 'react';
import PropTypes from 'prop-types';

import { useFormatMessage } from 'hooks';

const ErrorMessage = ({ text = '' }) => {
  const defaultText = useFormatMessage('ErrorMessage.defaultMessage');
  return <p className="error">{text || defaultText}</p>;
};

ErrorMessage.propTypes = {
  text: PropTypes.string,
};

export default ErrorMessage;
