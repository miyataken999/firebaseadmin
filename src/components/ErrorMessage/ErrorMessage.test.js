import React from 'react';

import ErrorMessage from '.';

describe('<ErrorMessage /> rendering', () => {
  it('should render without crashing', () => {
    const { component } = renderWithProviders(<ErrorMessage />)({});

    expect(component.asFragment()).toMatchSnapshot();
  });
});
