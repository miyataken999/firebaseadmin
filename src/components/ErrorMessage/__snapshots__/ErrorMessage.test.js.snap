// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> rendering should render without crashing 1`] = `
<DocumentFragment>
  <p
    class="error"
  >
    This field is required
  </p>
</DocumentFragment>
`;
